### DISCUSSION TITLE
Are there plans to add convective outlook and risk probabilities to the updated API?

### DISCUSSION THREAD

**Discussion by blakek** [weather-gov/api#586] [Q&A]
First, thanks for all the work y'all do!

I'm working on some projects that would benefit from data being added to the API:

- categorical convective outlooks for a location (e.g. 0=TSTM, 1=MRGL ... 5=HIGH)
- probabilities of different events (e.g. "Probability of Extreme Tornadoes" as a percentage)

For the categorical convective outlooks, it seems parsing outlook KMZ files may be the best current method (e.g. https://www.spc.noaa.gov/products/outlook/day1otlk_cat.kmz). For risk probabilities, I've used the NDFD endpoint in the past (e.g. https://graphical.weather.gov/xml/sample_products/browser_interface/ndfdXMLclient.php).

Are there plans to add these to api.weather.gov? If not, is there a better way to fetch these?

Thanks!

**Comment by netbymatt** [comment]
I completely agree that having both of these as part of one of the /gridpoints/<WFO>/<x,y> endpoints would be very helpful.

For now, if it helps and you're using Javascript I'm parsing the geoJson files provided by the SPC at these locations:
https://www.spc.noaa.gov/gis/

GeoJson was an intentional choice for my platform because I could easily parse and process it in the user's browser without needing a KMZ library to deal with unzipping and parsing that format.

You can view my parsing code which processes the user's location and polygons to see if there is a hit within any of them.
Getting and processing the geojson files:
https://github.com/netbymatt/temperature-express/blob/main/html/resources/js/spcoutlook.mjs
Point in polygon testing:
https://github.com/netbymatt/temperature-express/blob/main/html/resources/js/spcoutlook/polygon.mjs
